code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -1439,3 +1440,41 @@ nobiliumsuppression = INFINITY
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
 			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
 	return REACTING
+
+/datum/gas_reaction/miasma_decomp
+	priority = 0
+	name = "miasma decomposition"
+	id = "miasma_decomp"
+
+/datum/gas_reaction/miasma_decomp/init_reqs()
+	min_requirements = list(
+		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
+		/datum/gas/miasma = MINIMUM_MOLE_COUNT,
+		"TEMP" = 350
+	)
+
+/datum/gas_reaction/miasma_decomp/react(datum/gas_mixture/air, datum/holder)
+	var/energy_released = 0
+	var/old_heat_capacity = air.heat_capacity()
+	var/list/cached_gases = air.gases //this speeds things up because accessing datum vars is slow
+	var/temperature = air.temperature
+	var/burned_fuel = 0
+	burned_fuel = min((temperature-300)/10, cached_gases[/datum/gas/oxygen][MOLES], cached_gases[/datum/gas/miasma][MOLES])
```

Comment:
This is a lot, you could make an easy farm in lavaland with one t4 heater at 1300 K and you would be able to convert up to 100 moles per tick. All without much effort

Reply:
t4 parts aren't that easy to get your hands on really and even then you'd need to separate the co2.

Reply:
is not that hard to separate the CO2 and you can still do it automatically